app/services/rent_service.rb

```diff
@@ -0,0 +1,56 @@
+module RentService
+  def self.reservation(user, rent)
+    Rent.transaction do
+      rent.validate!
+      rent.user = user
+      rent.status = user.try(:member?) ? :confirmed : :unconfirmed
+      rent.save
+      if rent.council.present?
+        rent.overlap.each(&:overbook)
+      end
+      RentMailer.rent_email(rent).deliver_now
```

Comment:
@jforberg 
When sending confirmation emails like this, when should it be done?
I want to avoid using any `after_save` or such, but would this failing make the other actions be reversed?